frontend/src/stores/auth.js

```diff
@@ -3,8 +3,36 @@ import { defineStore } from 'pinia'
 
 import ApiService from '@/common/apiService'
 import AuthService from '@/common/authService'
+import FacilityService from '@/services/facilityService'
 import { useAppStore } from '@/stores/app'
-import { APPLICATION_INTAKE_TYPES } from '@/utils/constants'
+import { APPLICATION_INTAKE_TYPES, OFM_PROGRAM_CODES } from '@/utils/constants'
+
+/*
+  A facility can apply for a core application if it satisfies these conditions:
+  1. There is an Open Intake or the facility is in the allowed facility list of a Limited Intake
+  2. If the facility is in the CCOF program, their program start date must >= 1 year from the current date.
+*/
+function intakeWindowCheckForAddApplication(facility) {
+  const appStore = useAppStore()
+  const intakeWindowCheck = appStore.applicationIntakes?.some((intake) => {
+    const isWithinApplicationIntakeWindow = moment().isSameOrAfter(moment(intake.startDate)) && moment().isSameOrBefore(moment(intake.endDate))
+    const isOpenIntake = intake.type === APPLICATION_INTAKE_TYPES.OPEN_INTAKE
+    const limitedIntakeFacilities = intake?.facilities?.map((facility) => facility.facilityId)
+    return isWithinApplicationIntakeWindow && (isOpenIntake || limitedIntakeFacilities?.includes(facility.facilityId))
+  })
+  return intakeWindowCheck
+}
+
+async function ccofEnrolmentCheckForAddApplication(facility) {
+  try {
+    const facilityInfo = await FacilityService.getFacility(facility?.facilityId)
```

Comment:
Would it make more sense to have ofm_ccof_requirement/ccofOneYearEnrolment available as part of the ProviderProfile payload? It'll save a query for each facility.

Reply:
it's updated. thanks for your suggestions!